tests/js/spec/utils/discover/discoverQuery.spec.jsx

```diff
@@ -91,4 +91,70 @@ describe('DiscoverQuery', function () {
       })
     );
   });
+
+  it('parses string errors correctly', async function () {
+    MockApiClient.addMockResponse({
+      url: '/organizations/test-org/eventsv2/',
+      body: {
+        detail: 'Error Message',
+      },
+      statusCode: 400,
+    });
+
+    let errorValue;
+    render(
+      <DiscoverQuery
+        orgSlug="test-org"
+        api={api}
+        location={location}
+        eventView={eventView}
+        setError={e => (errorValue = e)}
+      >
+        {({isLoading}) => {
+          if (isLoading) {
+            return 'loading';
+          }
+          return null;
+        }}
+      </DiscoverQuery>
+    );
+    await tick();
```

Comment:
can we use `waitFor` or some async RTL query here?

Reply:
One day 😄  (I'll prob just convert the whole file at once in a future PR)